runner/arazzo_runner/extractor/openapi_extractor.py

```diff
@@ -239,6 +239,40 @@ def _resolve_schema_refs(
     return [_resolve_schema_refs(item, full_spec, stack, memo) for item in schema_part]
 
 
+def _extract_schema_of_content_encoding(body_content: dict[str, Any]) -> dict[str, Any]:
+    """
+    Extract schema from request/response body content, prioritizing JSON over form-encoded.
+
+    This function looks for supported content types in the body_content dictionary:
+    1. First tries to find any content type that starts with "application/json"
+    2. If no JSON content type found, looks for "application/x-www-form-urlencoded"
+    3. Returns the schema from the first matching content type, or None if none found
+
+    This handles content types with parameters like "application/json;q=0.7" by matching
+    the prefix rather than requiring exact key matches.
+
+    Args:
+        body_content: Dictionary mapping content types to their schemas
+
+    Returns:
+        The schema object from the first supported content type, or None if none found
+    """
+    content_type = next(
+        (key for key in body_content.keys() if key.startswith("application/json")), None
+    )
+    if not content_type:
+        content_type = next(
+            (
+                key
+                for key in body_content.keys()
+                if key.startswith("application/x-www-form-urlencoded")
+            ),
+            None,
+        )
+
+    return body_content.get(content_type, {}).get("schema")
```

Comment:
We're assuming that `schema` field is there and it's of proper type: https://github.com/jentic/arazzo-engine/pull/116/files#diff-450be3f98d26870bfc6769dfd12d2f9b170f42e06ca348a05e6d3844d464854bR242. But `MediaType.schema` is not required field so it can be `None`. 

Will the runner reject running the workflow/operation if the API Document is not valid? If not, we should probably handle these cases carefully.